test/faktory_worker/connection_manager_test.exs

```diff
@@ -61,6 +61,36 @@ defmodule FaktoryWorker.ConnectionManagerTest do
       assert result == "OK"
     end
 
+    test "should not retry a unique job that is already in the queue" do
+      connection_mox()
+
+      payload = %{
+        args: [1],
+        jid: "123456",
+        jobtype: "TestJob",
+        queue: "test_queue"
+      }
+
+      json = Jason.encode!(payload)
+      command = "PUSH #{json}\r\n"
+
+      expect(FaktoryWorker.SocketMock, :send, fn _, ^command ->
```

Comment:
I can't move the send mock because it depends on the `command` variable which depends on everything else above it. I could move the recv mock, but I like that it comes after the send mock since that's the order of the calls. Does that make sense?

Reply:
Ah right yeah, fair enough. How about moving the `connection_mox()` call down just so it's grouped together.